tests/query/processors/test_filter_in_select_optimizer.py

```diff
@@ -50,107 +59,241 @@ def subscriptable_reference(name: str, key: str) -> SubscriptableReference:
+                    "tags_raw", str(mql_context["indexer_mappings"]["status_code"])
+                ),
+                Literal(None, "200"),
+            ),
+            equals(
+                Column("_snuba_metric_id", None, "metric_id"),
                 Literal(None, 123456),
-            },
-            subscriptable_reference("tags_raw", "222222"): {
+            ),
+        ),
+        _and(
+            equals(
+                subscriptable_reference(
+                    "tags_raw", str(mql_context["indexer_mappings"]["status_code"])
+                ),
                 Literal(None, "200"),
-            },
-        },
+            ),
+            equals(
+                Column("_snuba_metric_id", None, "metric_id"),
+                Literal(None, 123456),
+            ),
+        ),
+    ),
+    "(sum(`d:transactions/duration@millisecond`) / sum(`d:transactions/duration@millisecond`)) + 100": _or(
+        equals(
+            Column("_snuba_metric_id", None, "metric_id"),
+            Literal(None, 123456),
+        ),
+        equals(
+            Column("_snuba_metric_id", None, "metric_id"),
+            Literal(None, 123456),
+        ),
     ),
-    (
-        "(sum(`d:transactions/duration@millisecond`) / sum(`d:transactions/duration@millisecond`)) + 100",
-        {
-            Column("_snuba_metric_id", None, "metric_id"): {
+    "sum(`d:transactions/duration@millisecond`) * 1000": equals(
+        Column("_snuba_metric_id", None, "metric_id"),
+        Literal(None, 123456),
+    ),
+    "1 + sum(`d:transactions/duration@millisecond`){status_code:200} / sum(`d:transactions/duration@millisecond`)": _or(
+        equals(
+            Column("_snuba_metric_id", None, "metric_id"),
+            Literal(None, 123456),
+        ),
+        _and(
+            equals(
+                subscriptable_reference(
+                    "tags_raw", str(mql_context["indexer_mappings"]["status_code"])
+                ),
+                Literal(None, "200"),
+            ),
+            equals(
+                Column("_snuba_metric_id", None, "metric_id"),
                 Literal(None, 123456),
-            },
-        },
+            ),
+        ),
     ),
-]
+}
 
 """ TESTING """
 
-optimizer = FilterInSelectOptimizer()
-
 
 @pytest.mark.parametrize(
-    "mql_query, expected_domain",
-    mql_test_cases,
+    "mql_query, expected_condition",
+    expected_optimize_condition.items(),
 )
-def test_get_domain_of_mql(mql_query: str, expected_domain: set[int]) -> None:
+def test_condition_generation(mql_query: str, expected_condition: FunctionCall) -> None:
     logical_query, _ = parse_mql_query(str(mql_query), mql_context, generic_metrics)
     assert isinstance(logical_query, Query)
-    res = optimizer.get_domain_of_mql_query(logical_query)
-    if res != expected_domain:
-        raise
-    assert res == expected_domain
+
+    opt = FilterInSelectOptimizer()
+    actual = opt.get_select_filter(logical_query)
```

Comment:
this test does not actually test that the query processor is applying the changes to the condition (which is its reason for being). The test should assert that behavior

Reply:
tests changed to test: input_query -> expected_output_query